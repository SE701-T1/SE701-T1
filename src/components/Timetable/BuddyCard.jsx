import * as React from 'react';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import PropTypes from 'prop-types';
import AccountCircleRoundedIcon from '@mui/icons-material/AccountCircleRounded';
import styles from './BuddyCard.module.css';

/**
 * This component is intended to be displayed inside the CoursesPageCard and displays the buddy name.
 * The component takes in two optional props. Conditional rendering of this component is done based
 * on whether or not the props are passed in.
 *
 */

export default function BuddyCard({ buddyName, courseName }) {
  if (!buddyName) {
    return (
      <Card className={styles.cardContainer}>
        <CardContent>
          <Typography variant="body">You dont have any buddies in this course yet</Typography>
        </CardContent>
        <CardActions>
          <Button size="Large" variant="outlined" color="inherit" className={styles.button}>
            Find a buddy
          </Button>
        </CardActions>
      </Card>
    );
  }

  return (
    <Card className={styles.cardContainer}>
      <CardContent className={styles.cardContent}>
        <AccountCircleRoundedIcon />
        <Typography variant="body">{buddyName}</Typography>
        <Typography variant="subtitle">Also takes {courseName}</Typography>
      </CardContent>
    </Card>
  );
}

BuddyCard.propTypes = {
  buddyName: PropTypes.string,
  courseName: PropTypes.string,
};

BuddyCard.defaultProps = {
  buddyName: '',
  courseName: '',
};
