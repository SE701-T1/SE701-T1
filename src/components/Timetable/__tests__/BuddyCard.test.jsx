import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import BuddyCard from '../BuddyCard';

describe('BuddyCard render with no props', () => {
  it('should display default text', () => {
    render(<BuddyCard />);

    expect(screen.getByText(/You dont have any buddies in this course yet/i)).toBeInTheDocument();
  });
  it('should display find buddy button', () => {
    render(<BuddyCard />);

    expect(screen.getByRole('button', { name: /Find a buddy/i })).toBeInTheDocument();
  });
});

describe('BuddyCard render with props', () => {
  it('should display buddy name', () => {
    render(<BuddyCard buddyName="Example Name" courseName="SE751" />);

    expect(screen.getByText(/Example Name/i)).toBeInTheDocument();
  });
  it('should display course name', () => {
    render(<BuddyCard buddyName="Example Name" courseName="SE751" />);

    expect(screen.getByText(/SE751/i)).toBeInTheDocument();
  });
});

it('produces correct snapshot of the BuddyCard', () => {
  const tree = renderer.create(<BuddyCard buddyName="Example Name" courseName="SE751" />).toJSON();
  expect(tree).toMatchSnapshot();
});
