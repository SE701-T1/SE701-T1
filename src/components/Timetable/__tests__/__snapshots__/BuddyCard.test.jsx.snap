// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`produces correct snapshot of the BuddyCard 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root cardContainer css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root cardContent css-46bh2p-MuiCardContent-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AccountCircleRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
      />
    </svg>
    <span
      className="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
    >
      Example Name
    </span>
    <span
      className="MuiTypography-root MuiTypography-subtitle css-37jdci-MuiTypography-root"
    >
      Also takes 
      SE751
    </span>
  </div>
</div>
`;
